### PR TITLE
Navigation fix

### PR DESCRIPTION
**Pull request by annarog**
Fix to the issue with switching tabs and refreshing content
#98

### CODE REVIEW

File: angular-client/src/app/components/users/register-user/register-user.component.ts [hunk 1]
```diff
@@ -23,6 +26,7 @@ export class RegisterUserComponent implements OnInit {
   msg = '';
```
Comment (Ldhatter) [inline angular-client/src/app/components/users/register-user/register-user.component.ts]: is this used anymore?

File: angular-client/src/app/components/users/register-user/register-user.component.ts [hunk 2]
```diff
@@ -37,15 +41,19 @@ export class RegisterUserComponent implements OnInit {
...
+        });
       } else {
-        this.authenticationService.loggedIn = false;
+        this.alertModal('Could not add new user.').subscribe(() => {
+          console.log(data);
+        });
+
         this.loading = false;
         this.error = true;
         this.msg = '';
```
Comment (Ldhatter) [inline angular-client/src/app/components/users/register-user/register-user.component.ts]: same here? is this used?

File: angular-client/src/app/components/participants/add-participant/add-participant.component.ts [hunk 3]
```diff
@@ -97,10 +96,15 @@ export class AddParticipantComponent implements OnInit {
   onSubmit() {
     this.participantService.save(this.form.value)
       .subscribe(data => {
-        if (data.hasOwnProperty('errmsg')) {
-          this.alertModal('Could not add new participant.');
+        if (data.hasOwnProperty('errors')) {
+          this.alertModal('Could not add new participant.').subscribe( () => {
+            console.log(data);
```
Comment (Ldhatter) [inline angular-client/src/app/components/participants/add-participant/add-participant.component.ts]: could remove

File: angular-client/src/app/components/phonelog/add-phonelog/add-phonelog.component.ts [hunk 4]
```diff
@@ -88,10 +85,15 @@ export class AddPhonelogComponent implements OnInit {
 
     this.phonelogService.save(formModel)
       .subscribe(data => {
-        if (data.hasOwnProperty('errmsg')) {
-          this.alertModal('Could not add new phonelog entry.');
+        if (data.hasOwnProperty('errors')) {
+          this.alertModal('Could not add new phonelog entry.').subscribe( () => {
+            console.log(data);
```
Comment (Ldhatter) [inline angular-client/src/app/components/phonelog/add-phonelog/add-phonelog.component.ts]: could remove

File: angular-client/src/app/components/resources/add-resource/add-resource.component.ts [hunk 5]
```diff
@@ -68,11 +67,15 @@ export class AddResourceComponent implements OnInit {
   submit() {
     this.resourceService.save(this.form.value['kind'].toLowerCase(), this.form.value)
       .subscribe(data => {
-        if (data.hasOwnProperty('errmsg')) {
-          this.alertModal('Could not add new resource.');
+        if (data.hasOwnProperty('errors')) {
+          this.alertModal('Could not add new resource.').subscribe( () => {
+            console.log(data);
```
Comment (Ldhatter) [inline angular-client/src/app/components/resources/add-resource/add-resource.component.ts]: could remove

File: angular-client/src/app/components/users/register-user/register-user.component.ts [hunk 6]
```diff
@@ -37,15 +41,19 @@ export class RegisterUserComponent implements OnInit {
...
 
       if (!data.error) {
-        this.msg = data.msg;
+        this.alertModal('New user successfully created.').subscribe(() => {
+          this.router.navigateByUrl('/dashboard/users');
+        });
       } else {
-        this.authenticationService.loggedIn = false;
+        this.alertModal('Could not add new user.').subscribe(() => {
+          console.log(data);
```
Comment (Ldhatter) [inline angular-client/src/app/components/users/register-user/register-user.component.ts]: could remove

File: angular-client/src/app/components/users/register-user/register-user.component.ts (hunk 1 above)
Comment (annarog) [reply angular-client/src/app/components/users/register-user/register-user.component.ts]: Yep, it's used to display an error message

File: angular-client/src/app/components/users/register-user/register-user.component.ts (hunk 2 above)
Comment (annarog) [reply angular-client/src/app/components/users/register-user/register-user.component.ts]: Yep it's used a couple lines down

**Pull request merged**